lib/mappers/wisconsin_mapper.py

```diff
@@ -8,42 +10,129 @@ def __init__(self, provider_data):
+        if exists(self.provider_data, 'dataProvider'):
+            self.mapped_data.update(
+                {'dataProvider': getprop(self.provider_data, 'dataProvider')})
+
 
     def map_is_shown_at(self):
         """//edm:isShownAt -> .isShownAt (required property)"""
-        self.mapped_data.update(
-            {'isShownAt': getprop(self.provider_data, 'isShownAt')})
+        if exists(self.provider_data, 'isShownAt'):
+            self.mapped_data.update(
+                {'isShownAt': getprop(self.provider_data, 'isShownAt')})
+
+    def map_object(self):
+        if exists(self.provider_data, 'preview'):
+            self.mapped_data.update(
+                {'object': getprop(self.provider_data, 'preview')})
 
     def map_collection(self):
         """//dct:isPartOf -> sourceResource.collection"""
+        collection = []
         if exists(self.provider_data, 'isPartOf'):
-            self.update_source_resource(
-                {'collection': iterify(getprop(self.provider_data,
-                                               'isPartOf'))})
+            for coll in iterify(getprop(self.provider_data, 'isPartOf')):
+                collection.append({"title": coll})
+        if collection:
+            self.update_source_resource({ "collection": collection })
 
     def map_format(self):
         """//dc:format or //dc:medium -> .sourceResource.format"""
-        self.source_resource_prop_from_many('format', ('medium', 'format'))
-
-    def map_spec_type(self):
-        """//dc:format -> .sourceResource.specType (a.k.a. genre in MAPv4)"""
-        if exists(self.provider_data, 'format'):
-            self.update_source_resource(
-                {'specType': iterify(getprop(self.provider_data, 'format'))})
+        format = []
+        self.extend_collection(format, 'medium')
+        self.extend_collection(format, 'format')
+        if format:
+            self.update_source_resource({'format': format})
 
     def map_identifier(self):
         """//dc:identifier -> .sourceResource.identifier"""
-        self.source_resource_prop_to_prop('identifier')
+        if exists(self.provider_data, "dc:identifier"):
+            identifiers = []
+            for identifier in iterify(self.provider_data.get("identifier")):
+                if isinstance(identifier, dict) and "#text" in identifier:
+                    identifiers.append(identifier["#text"])
+                elif isinstance(identifier, basestring):
+                    identifiers.append(identifier)
+            identifiers = filter(None, identifiers)
+            if identifiers:
+                self.update_source_resource({"identifier": identifiers})
+
+    def map_language(self):
+        lang = []
+        if exists(self.provider_data, "language"):
+            for s in iterify(getprop(self.provider_data, "language")):
+                if isinstance(s, dict):
+                    lang.append(s.get("#text"))
+                else:
+                    lang.append(s)
```

Comment:
As above, I think that this could also be a list comprehension with `textnode.textnode()`